src/pixel/state.ts

```diff
@@ -24,7 +25,7 @@ const paramExtractors: ((state: State) => [string, string][])[] = [
   ifDefined('eventSource', source => asParamOrEmpty('se', source, (s) => base64UrlEncode(JSON.stringify(s, replacer)))),
   ifDefined('liveConnectId', fpc => asStringParam('duid', fpc)),
   ifDefined('trackerName', tn => asStringParam('tna', tn)),
-  ifDefined('pageUrl', purl => asStringParam('pu', purl)),
+  state => ifDefined('pageUrl', purl => asStringParam('pu', collectUrl(purl, state)))(state),
```

Comment:
Would it be possible to do the URL parsing only if it is required. This way we would not make the script slower for customers that do not use that feature.